src/segmented-control/internal-segmented-control.tsx

```diff
@@ -49,7 +51,7 @@ export default function InternalSegmentedControl({
       {options &&
         options.map((option: SegmentedControlProps.Option, index) => {
           const isActive = selectedId === option.id;
-          const enabledSegmentIndex = enabledSegments.indexOf(option);
+          const enabledSegmentIndex = focusableSegments.indexOf(option);
```

Comment:
Guess it would be good to rename `enabledSegmentIndex` to `focusableSegemtIndex` too :)